### ISSUE TITLE
Constructors for immutable unboxed tuple vectors are not exported

### ISSUE THREAD

**Issue by ruifengx** [haskell/vector#504]
See the following export list for `Data.Vector.Unboxed`:

https://github.com/haskell/vector/blob/eb605264be1811472b447b57627b2074e63614fe/vector/src/Data/Vector/Unboxed.hs#L61-L63

All the constructors for the `MVector` family are exported, which includes constructors for mutable unboxed tuple vectors (e.g., `MV_2`). However, for immutable unboxed tuple vectors, the constructors are private.

Currently, to construct an unboxed tuple vector from the two underlying parallel vector without copying, one must go through `MVector` and `unsafeFreeze`, which seems a bit weird to me.
```haskell
import Data.Vector.Unboxed qualified as U
pairUnboxed :: U.Vector a -> U.Vector b -> U.Vector (a, b)
pairUnboxed xs ys = do
    xs' <- unsafeThaw xs
    ys' <- unsafeThaw ys
    let len = U.length xs
    unsafeFreeze (MV_2 len xs' ys')
```

**Comment by toyboot4e** [comment]
Related issues: #49, #492.

**Comment by ruifengx** [comment]
Indeed, this seems like a duplicate of #492, then. I guess the only new point is that not exporting `V_2` does not actually prevent the user from achieving the same functionality, just with more boilerplate. Maybe I should close this one in favour of #492?

**Comment by Shimuuar** [comment]
In fact `zip/unzip`, `zip3/unzip3` etc exported from `D.V.Unboxed` and  will grant access to underlying vectors. This is just poorly documented. Basically this is a documentation problem

**Comment by ruifengx** [comment]
Thanks! Somehow I did not think this way. I must have thought those functions are just aliases to the ones in `D.V.Generic` (for fusion, of course), and I did not find their implementations on Hackage. Checking their implementations in the included file `unboxed-tuple-instances`, it is indeed just what I was looking for:
https://github.com/haskell/vector/blob/eb605264be1811472b447b57627b2074e63614fe/vector/internal/unbox-tuple-instances#L123-L130

Actually, in [their documentation](https://hackage.haskell.org/package/vector-0.13.1.0/docs/Data-Vector-Unboxed.html#v:zip) they are clearly marked `O(1)`. Maybe it can be improved in discoverability by mentioning this fact at the very top where representations of unboxed vectors are discussed.
